services/crates/crates-msrv.service.js

```diff
@@ -0,0 +1,78 @@
+import { NotFound, pathParams } from '../index.js'
+import {
+  BaseCratesService,
+  description as cratesIoDescription,
+} from './crates-base.js'
+
+const description = `
+${cratesIoDescription}
+
+MSRV is a crate's minimum suppported rust version,
+the oldest version of Rust supported by the crate.
+See the [Cargo Book](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field)
+for more info.
+`
+
+export default class CratesMSRV extends BaseCratesService {
+  static category = 'platform-support'
+  static route = {
+    base: 'crates/msrv',
+    pattern: ':crate/:version?',
+  }
+
+  static openApi = {
+    '/crates/msrv/{crate}': {
+      get: {
+        summary: 'Crates.io MSRV',
+        description,
+        parameters: pathParams({
+          name: 'crate',
+          example: 'serde',
+        }),
+      },
+    },
+    '/crates/msrv/{crate}/{version}': {
+      get: {
+        summary: 'Crates.io MSRV (version)',
+        description,
+        parameters: pathParams(
+          {
+            name: 'crate',
+            example: 'serde',
+          },
+          {
+            name: 'version',
+            example: '1.0.194',
+          },
+        ),
+      },
+    },
+  }
+
+  static defaultBadgeData = { label: 'MSRV', color: 'blue' }
+
+  static render({ msrv: message }) {
+    return { message }
+  }
+
+  static transform({ errors, version, versions }) {
+    // crates.io returns a 200 response with an errors object in
+    // error scenarios, e.g. https://crates.io/api/v1/crates/libc/0.1
+    if (errors) {
+      throw new NotFound({ prettyMessage: errors[0].detail })
+    }
+
+    const msrv = version ? version.rust_version : versions[0].rust_version
+    if (!msrv) {
+      throw new NotFound({ prettyMessage: 'unknown' })
+    }
+
+    return { msrv: `${msrv}+` }
```

Comment:
Looking at crates.io, they present this as just the version number (without a plus). I'd be inclined to match the upstream registry on this.

Reply:
The MSRV indicates not a specific version requirement, but a "this version or later" requirement (hence the 'minimum supported' in 'minimum supported rust version').

I've opened https://github.com/rust-lang/crates.io/pull/7888 to possibly change this on the crates.io side. I'm willing to match whatever the crates.io team decides.

Reply:
update: crates.io team thinks the plus is unnecessary
https://github.com/rust-lang/crates.io/pull/7888#issuecomment-1879820085